### PR TITLE
Add tests for sycl::errc

### PR DESCRIPTION
**Pull request by kopylovanat**

### CODE REVIEW

File: tests/exceptions/exceptions.h [hunk 1]
```diff
@@ -16,22 +16,60 @@
...
-    sycl::errc::build,
-    sycl::errc::invalid,
-    sycl::errc::memory_allocation,
-    sycl::errc::platform,
-    sycl::errc::profiling,
-    sycl::errc::feature_not_supported,
-    sycl::errc::kernel_not_supported,
-    sycl::errc::backend_mismatch};
+const std::vector<sycl::errc>& get_err_codes() {
+  static const std::vector<sycl::errc> all_err_codes{
```
Comment (bader) [inline tests/exceptions/exceptions.h:20]: I suppose using dynamic size array is not necessary. `std::array` seems to fit better here.

**Review by bader — APPROVED** [review]

File: tests/exceptions/exceptions.h (hunk 1 above)
Comment (kopylovanat) [reply tests/exceptions/exceptions.h]: Replaced std::vector with std::array

File: tests/exceptions/exceptions_error_code.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  void get_info(test_base::info &out) const override {
+    set_test_info(out, TOSTRING(TEST_NAME), TEST_FILE);
+  }
+
+  /** execute the test
+   */
+  void run(util::logger &log) override {
+    try {
+      const auto all_err_codes {get_err_codes()};
+      for (auto &err_c : all_err_codes) {
```
Comment (keryell) [inline tests/exceptions/exceptions_error_code.cpp:49]:
```suggestion
      for (const auto &err_c : get_err_codes()) {
```

File: tests/exceptions/exceptions_make_error_code.cpp [hunk 3]
```diff
@@ -96,6 +96,7 @@ class TEST_NAME : public util::test_base {
    */
   void run(util::logger &log) override {
     try {
+      const auto all_err_codes {get_err_codes()};
```
Comment (keryell) [inline tests/exceptions/exceptions_make_error_code.cpp:99]: Fuse this in the loop after?

**Review by keryell — CHANGES_REQUESTED** [review]

File: tests/exceptions/exceptions.h [hunk 4]
```diff
@@ -16,22 +16,60 @@
...
-    sycl::errc::build,
-    sycl::errc::invalid,
-    sycl::errc::memory_allocation,
-    sycl::errc::platform,
-    sycl::errc::profiling,
-    sycl::errc::feature_not_supported,
-    sycl::errc::kernel_not_supported,
-    sycl::errc::backend_mismatch};
+const std::array<sycl::errc, 15>& get_err_codes() {
+  static const std::array<sycl::errc, 15> all_err_codes{
```
[the remark this answers is not in the document]
Comment (kopylovanat) [reply tests/exceptions/exceptions.h]: Done.

File: tests/exceptions/exceptions_error_code.cpp (hunk 2 above)
Comment (kopylovanat) [reply tests/exceptions/exceptions_error_code.cpp]: Done.

File: tests/exceptions/exceptions_make_error_code.cpp (hunk 3 above)
Comment (kopylovanat) [reply tests/exceptions/exceptions_make_error_code.cpp]: Moved get_err_codes() to loop.

**Review by ProGTX — APPROVED** [review]

**Review by keryell — APPROVED** [review]
Good!

**Pull request merged**